Source/Core/Core.swift

```diff
@@ -481,11 +481,36 @@ public class FormViewController : UIViewController, FormViewControllerProtocol {
     
     public override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
-        if let selectedIndexPath = tableView?.indexPathForSelectedRow {
-            tableView?.reloadRowsAtIndexPaths([selectedIndexPath], withRowAnimation: .None)
-            tableView?.selectRowAtIndexPath(selectedIndexPath, animated: false, scrollPosition: .None)
-            tableView?.deselectRowAtIndexPath(selectedIndexPath, animated: true)
+
+        let selectedIndexPaths = tableView?.indexPathsForSelectedRows ?? []
+        tableView?.reloadRowsAtIndexPaths(selectedIndexPaths, withRowAnimation: .None)
+        selectedIndexPaths.forEach {
+            tableView?.selectRowAtIndexPath($0, animated: false, scrollPosition: .None)
+        }
+
+        let deselectionAnimation = { (context: UIViewControllerTransitionCoordinatorContext) in
```

Comment:
Fair enough. A good rule of thumb is that if it's an animation block, and it's going to run once, then it's fine to not use `[weak self]`. If it's a repeating animation, more care must be taken, because the animation system will retain the block indefinitely, so you can't rely on canceling the animation in `deinit`.